src/gfnff/gfnff_eg.f90

```diff
@@ -136,7 +142,13 @@ subroutine gfnff_eg(env,pr,n,ichrg,at,xyz,makeq,g,etot,res_gff, &
      &         eeqtmp(2,n*(n+1)/2),d3list(2,n*(n+1)/2),dcn(3,n,n),cn(n), &
      &         hb_dcn(3,n,n),hb_cn(n))
 
-      if (pr) call timer%new(10 + count([allocated(solvation)]),.false.)
+      if (pr) then   
+          call timer%new(10 + count([allocated(solvation)]),.false.)
+      else if (present(minpr) .and. minpr) then
```

Comment:
You cannot check for presence of an optional argument and use the argument in the same condition as the argument might not be present and the order of evaluation is not guaranteed.